tests/components/watts/test_init.py

```diff
@@ -0,0 +1,147 @@
+"""Test the Watts Vision integration initialization."""
+
+from unittest.mock import AsyncMock, patch
+
+from aiohttp import ClientError, ClientResponseError
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry
+
+
+async def test_setup_entry_success(
+    hass: HomeAssistant,
+    mock_watts_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test successful setup and unload of entry."""
+    await setup_integration(hass, mock_config_entry, mock_watts_client)
+
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+
+    mock_watts_client.discover_devices.assert_called_once()
+
+    unload_result = await hass.config_entries.async_unload(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert unload_result is True
+    assert mock_config_entry.state is ConfigEntryState.NOT_LOADED
+
+
+async def test_setup_entry_auth_failed(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setup with authentication failure."""
+
+    mock_config_entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation"
+        ) as mock_get_implementation,
+        patch(
+            "homeassistant.helpers.config_entry_oauth2_flow.OAuth2Session"
+        ) as mock_session,
```

Comment:
we can do without patching these I believe, I would assume we can have the aiohttp mock like in the config flow